doc/manual/src/glossary.md

```diff
@@ -115,14 +115,14 @@
     then be built.
 
   - [reference]{#gloss-reference}\
-    A store path `P` is said to have a reference to a store path `Q` if
-    the store object at `P` contains the path `Q` somewhere. The
-    *references* of a store path are the set of store paths to which it
-    has a reference.
+    A [store object] `O` is said to have a *reference* to a store object `P` if the [store path] of `P` appears in the contents of `O`.
```

Comment:
This is not quite correct. References are not store objects but store paths.

There is no such thing as "the store path of P" since store objects don't have a uniquely determined store path. (E.g. there can be multiple store paths with the same contents.)

Reply:
I would say store paths are you unique, and those are not the same store object. I don't think I am "right' and you are "wrong" though, I think these are both valid opinions as of today, and we decide which one we want to be correct and formalize it.

In other words, this is an interesting question, not something to be shoved underneath the rug!

Reply:
Here is one way to fix it
```suggestion
    A [store object] `O` is said to have a *reference* to a store object `P` if a [store path] of `P` appears in the contents of `O`.
```

Reply:
here is another way
```suggestion
    Store objects can *refer* to other store objects by [store path].
    A [store object] `O` is said to have a *reference* `P` to some other store object if `P` appears in the contents of `O`.
```

Reply:
These are valid with both interpretations.

I do have a preference for not saying "store path of" even if the paths are unique, on the "things shouldn't know their own name" principle.

Reply:
> store objects don't have a uniquely determined store path

> I would say store paths are unique, and [two store objects with same contents but different paths] are not the same store object.

@edolstra @Ericson2314 Good points, both. Let's determine and document the mental model we want users to apply. Intuitively I'm inclined towards the latter, because input addressing is a concept that's easy to explain and understand. It's the derivations that produce store objects and define their store paths. That different derivations may produce the same store object is incidental. Deduplicating store objects with the same contents and aliasing those under different paths appears to be an implementation detail that is not even necessarily a concern of Nix but that of the underlying persistence layer.

Until this is resolved, I would go with the following, because it's brief and not wrong w.r.t. the current implementation:

```
A [store object] `O` is said to have a *reference* to a store object `P` if a [store path] to `P` appears in the contents of `O`.
```